src/lf_apply.rs

```diff
@@ -377,7 +394,7 @@ unsafe fn filter_plane_cols_y<BD: BitDepth>(
                 dst.offset((x * 4) as isize).cast(),
                 ls,
                 hmask.as_mut_ptr(),
-                lvl[x as usize][0..].as_ptr() as *const [u8; 4],
+                &lvl[x as usize],
```

Comment:
If enforcing a minimum slice size, this would be something like `lvl[x as usize..][..((endy4 - starty4 - 1) as isize * b4_stride + 1) as usize].as_ptr()`. However, index `-1` of such as slice can also be read, so more work on slice definitions will be needed. But that's probably outside the scope of this PR

Reply:
I was hoping to structure things as, at this point in the code, we don't care about checking the length, as it is up to the inner `fn` to correctly bounds check when reading elements.  That's of course not true for the asm `fn`s, though it will be eventually true (once we make them safe) for the C/Rust fallbacks.  If we do check the length eagerly here, the only real safety that gives us in documentation, that is, if we check that the asm `fn` claims it will access X elements and here we check for X elements, we check that the actual length matches the docs, but it doesn't enforce anything on the asm.

Reply:
We can't change the existing `fn` ptr args without modifying the asm, which we don't want to do.  But what we can do is is pass extra args that asm ignores (which we already do for passing `bitdepth_max: c_int` for 8bpc `fn`s).  So we can split a slice into the ptr and len, pass the len as an extra arg, and then recreate the slice in the fallback C/Rust `fn`.  Where it will be technically `unsafe`, but quite safe in practice.

As for asm `fn`s accessing more memory locations than the fallbacks, we could eagerly checks the bounds, but it's still ultimately up to the asm not to access any more than that.  It's always going to be inherently very `unsafe`.